### PR TITLE
[W6.2i][F10-1]Ng Tek In 

### PR DESCRIPTION
**Pull request by 0WN463**

### CODE REVIEW

**Comment by 0WN463** [comment]
NOTE: I nuclear destroyed this fork  (under nusCS2113-AY1819S1), to create a new fork (under CS2113-AY1819S1-F10-1).
The push request still seems valid, but beware of any problems that might surface due to this

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -55,4 +55,15 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+
+    @Override
```
Comment (stephlewyh) [inline src/seedu/addressbook/data/person/Address.java:59]: You may apply a static modifier to `getPrintableString(Printable printables)`. A static method invoked without the need for creating an instance of a class, and the method is stored in one memory address.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -55,4 +55,15 @@ public int hashCode() {
...
+    public String getPrintableString(boolean showPrivate) {
+        if (isPrivate()) {
+            if (showPrivate)
+                return "{private Address: " + value + "}";
```
Comment (stephlewyh) [inline src/seedu/addressbook/data/person/Address.java:63]: You can consider formatting strings `Formatter` from java.util.Formatter.

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 3]
```diff
@@ -37,50 +37,32 @@ default boolean isSameStateAs(ReadOnlyPerson other) {
...
-        if (getEmail().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getEmail())
-                .append(" Address: ");
-        if (getAddress().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getAddress())
+        final String stringChain = textFormatter.getPrintableString(true, getName(), getPhone(), getEmail(), getAddress());
```
Comment (stephlewyh) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java:42]: You invoked the newly implemented `getPrintableString(..)` - good job!

**Pull request closed without merging**